File: Week 2 - 12-20-2021/whileloopDemo2.py
# while loops can also be used to validate input
repeat = int(input("Enter a value that is between 1 and 3:" ))

# when validating, use the while loop to check for bad data
# in this case, only loop when the value entered is less than 1 or greater than 3
while repeat < 1 or repeat > 3:
    # display an error message
    print("\nERROR: Number is out of range, try again. \n")
    # get another value of repeat
    repeat = int(input("Enter a value that is between 1 and 3:" ))

# once valid data is entered, execute the loop
while repeat > 0:
    # checks to see if the value of repeat is greater than 0
    print("This is a python loop")
    repeat -= 1

File: Week 4 - 01-10-2022/whileLoopDemo2.cpp
#include <iostream>
using namespace std;

int main()
{
	int repeat;
	
	// while loops can be used to validate input
	cout << "Enter a number between 1 and 5: ";
	cin >> repeat;
	
	// when validating, always check for bad data
	// in this case, we only loop if the value is less than 1 or greater than 5
	while(repeat < 1 or repeat > 5)
	{
		// display an error message, then get another input
		cout << "\nERROR - NUMBER ENTERED IS NOT BETWEEN 1 AND 5. TRY AGAIN! \n";
		cout << "Enter a number between 1 and 5: ";
		cin >> repeat;
	}
	
	// once valid data is entered, display the message
	while(repeat > 0)
	{
		cout << "This is C++" << endl;
		// update the condition by changing the value of repeat (subtract 1 from repeat)
		repeat--;
	}
}

